contrib/qat/compression/qatzstd/compressor/source/config.cc

```diff
@@ -0,0 +1,97 @@
+#include "contrib/qat/compression/qatzstd/compressor/source/config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Compression {
+namespace Qatzstd {
+namespace Compressor {
+
+QatzstdCompressorFactory::QatzstdCompressorFactory(
+    const envoy::extensions::compression::qatzstd::compressor::v3alpha::Qatzstd& qatzstd,
+    Event::Dispatcher& dispatcher, Api::Api& api, ThreadLocal::SlotAllocator& tls)
+    : compression_level_(
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(qatzstd, compression_level, ZSTD_CLEVEL_DEFAULT)),
+      enable_checksum_(qatzstd.enable_checksum()), strategy_(qatzstd.strategy()),
+      chunk_size_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(qatzstd, chunk_size, ZSTD_CStreamOutSize())),
+      enable_qat_zstd_(qatzstd.enable_qat_zstd()),
+      qat_zstd_fallback_threshold_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(
+          qatzstd, qat_zstd_fallback_threshold, DefaultQatZstdFallbackThreshold)),
+      tls_slot_(nullptr) {
+  if (qatzstd.has_dictionary()) {
+    Protobuf::RepeatedPtrField<envoy::config::core::v3::DataSource> dictionaries;
+    dictionaries.Add()->CopyFrom(qatzstd.dictionary());
+    cdict_manager_ = std::make_unique<ZstdCDictManager>(
+        dictionaries, dispatcher, api, tls, true,
+        [this](const void* dict_buffer, size_t dict_size) -> ZSTD_CDict* {
+          return ZSTD_createCDict(dict_buffer, dict_size, compression_level_);
+        });
+  }
+  if (enable_qat_zstd_) {
+    tls_slot_ = ThreadLocal::TypedSlot<QatzstdThreadLocal>::makeUnique(tls);
+    tls_slot_->set([](Event::Dispatcher&) { return std::make_shared<QatzstdThreadLocal>(); });
+  }
+}
+
+QatzstdCompressorFactory::QatzstdThreadLocal::QatzstdThreadLocal()
+    : initialized_(false), sequenceProducerState_(nullptr) {}
+
+QatzstdCompressorFactory::QatzstdThreadLocal::~QatzstdThreadLocal() {
+  if (initialized_) {
+    /* Free sequence producer state */
+    QZSTD_freeSeqProdState(sequenceProducerState_);
+    /* Stop QAT device, please call this function when
+    you won't use QAT anymore or before the process exits */
+    QZSTD_stopQatDevice();
+  }
+}
+
+void* QatzstdCompressorFactory::QatzstdThreadLocal::GetQATSession() {
+  // The session must be initialized only once in every worker thread.
+  if (!initialized_) {
+
+    int status = QZSTD_startQatDevice();
+    // RELEASE_ASSERT(status == QZSTD_OK, "failed to initialize hardware");
+    if (status != QZSTD_OK) {
+      ENVOY_LOG(warn, "Failed to initialize qat hardware");
+    } else {
+      ENVOY_LOG(debug, "Initialize qat hardware successful");
+    }
+    sequenceProducerState_ = QZSTD_createSeqProdState();
+    initialized_ = true;
+  }
+
+  return sequenceProducerState_;
+}
+
+Envoy::Compression::Compressor::CompressorPtr QatzstdCompressorFactory::createCompressor() {
+  if (enable_qat_zstd_) {
```

Comment:
This if-else branch seems useless. For the last parameter, you can just write it as `enable_qat_zstd_? tls_slot_->get()->GetQATSession(): nullptr`. But at first, I feel the `enable_qat_zstd` seems useless, I already comment it in the api.

Reply:
Done, thanks!